core/server/serve.go

```diff
@@ -147,11 +152,11 @@ func (rt *RoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 		}
 	}
 
-	//Define 5 second timeout
+	//Use Proxy Read Timeout
 	transport := &http.Transport{
 		DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
 			return (&net.Dialer{
-				Timeout: 5 * time.Second,
+				Timeout: time.Duration(domains.Config.Proxy.Timeout.Read).Abs() * time.Second,

```

Comment:
Out of curiosity, why are you using time.Duration here, instead of just the value like "5" directly?